Storage: Add Dell PowerFlex SDC operation mode (#13660)
canonical/lxd: doc/reference/storage_powerflex.md

[Dell PowerFlex](https://www.dell.com/en-us/shop/powerflex/sf/powerflex) is a software-defined storage solution from [Dell Technologies](https://www.dell.com/). Among other things it offers the consumption of redundant block storage across the network.

> LXD offers access to PowerFlex storage clusters using both transport protocols NVMe/TCP and Dell's Storage Data Client (SDC).

Reviewer: ```suggestion
LXD offers access to PowerFlex storage clusters using either NVMe/TCP or Dell's Storage Data Client (SDC) as the transport protocol.
```
We could also leave out "as the transport protocol" if that's misleading for SDC.
The current wording implies that you can use both at the same time, but you probably can't?

Author: Good point, you have to decide for either one of them when creating the storage pool. Fixed.

In addition, PowerFlex offers copy-on-write snapshots, thin provisioning and other features.

To use PowerFlex with NVMe/TCP, make sure you have the required kernel modules installed on your host system.
On Ubuntu these are `nvme_fabrics` and `nvme_tcp`, which come bundled in the `linux-modules-extra-$(uname -r)` package.
LXD takes care of connecting to the respective subsystem.

> When using the SDC, LXD requires it to be connected already to the Dell Metadata Manager (MDM) beforehand.
> As LXD doesn't setup the SDC, follow the official guides from Dell for configuration details.

Reviewer: ```suggestion
When using the SDC, LXD requires it to already be connected to the Dell Metadata Manager (MDM) beforehand.
As LXD doesn't set up the SDC, follow the official guides from Dell for configuration details.
```

Author: Thanks fixed.

LXD creates its volumes in those storage pools.

You can take a snapshot of any volume in PowerFlex, which will create an independent copy of the parent volume.
PowerFlex volumes get added as a drive to the respective LXD host the volume got mapped to.
In case of NVMe/TCP, the LXD host connects to one or multiple NVMe {abbr}`SDT (storage data targets)` provided by PowerFlex.
Those SDT run as components on the PowerFlex storage layer.

> In case of SDC, the LXD hosts don't setup any connection by themselves.

Reviewer: ```suggestion
In case of SDC, the LXD hosts don't set up the connection themselves.
```

Author: Fixed.
